### PR TITLE
Refactor BlockStoreLocation and simplify Allocator implementation

### PR DESCRIPTION
**Pull request by rongrong**
### What changes are proposed in this pull request?

Refactor BlockStoreLocation and make location checking more efficient.
Also removing deprecated code and delete EvictorUtils.

### CODE REVIEW

File: core/common/src/main/java/alluxio/worker/block/BlockStoreLocation.java [hunk 1]
```diff
@@ -154,7 +172,7 @@ public String mediumType() {
    * @return true if location is {@link #ANY_MEDIUM}
```
Comment (beinan) [inline core/common/src/main/java/alluxio/worker/block/BlockStoreLocation.java:172]: modify the java doc to match the new logic?

File: core/common/src/main/java/alluxio/worker/block/BlockStoreLocation.java [hunk 2]
```diff
@@ -112,13 +112,22 @@ public String tierAlias() {
     return mTierAlias;
   }
 
+  /**
+   * Check whether the location has no restriction, i.e., {@link #ANY_TIER_LOCATION}.
+   *
+   * @return true if location is equal to {@link #ANY_TIER_LOCATION}
```
Comment (beinan) [inline core/common/src/main/java/alluxio/worker/block/BlockStoreLocation.java:118]: Looks like we hide the constant of ANY_TIER, ANY_DIR and ANY_MEDIUM behind these public methods, so can we make the constant be private?

**Review by beinan — COMMENTED** [review]
just a couple of very minor issues

File: core/common/src/main/java/alluxio/worker/block/BlockStoreLocation.java (hunk 1 above)
Comment (rongrong) [reply core/common/src/main/java/alluxio/worker/block/BlockStoreLocation.java:172]: The logic is the same. `ANY_MEDIUM` is an empty string.

File: core/common/src/main/java/alluxio/worker/block/BlockStoreLocation.java (hunk 2 above)
Comment (rongrong) [reply core/common/src/main/java/alluxio/worker/block/BlockStoreLocation.java:118]: Made ANY_DIR and ANY_MEDIUM private. ANY_TIER is still used as a key in a map. The code looks hacky but will take more effort to clean up. Might do that later.

File: core/common/src/main/java/alluxio/worker/block/BlockStoreLocation.java (hunk 1 above)
Comment (beinan) [reply core/common/src/main/java/alluxio/worker/block/BlockStoreLocation.java:172]: got it, sounds good to me

**Review by beinan — APPROVED** [review]
LGTM

**Comment by beinan** [comment]
alluxio-bot, merge this please.

**Pull request merged**